--- a/src/libkrun/src/lib.rs
+++ b/src/libkrun/src/lib.rs
@@ -74,13 +74,13 @@
 
 // krunfw library name for each context
 #[cfg(all(target_os = "linux", not(feature = "tee")))]
-const KRUNFW_NAME: &str = "libkrunfw.so.5";
+const KRUNFW_NAME: &str = "$ORIGIN/libkrunfw.so.5";
 #[cfg(all(target_os = "linux", feature = "amd-sev"))]
-const KRUNFW_NAME: &str = "libkrunfw-sev.so.5";
+const KRUNFW_NAME: &str = "$ORIGIN/libkrunfw-sev.so.5";
 #[cfg(all(target_os = "linux", feature = "tdx"))]
-const KRUNFW_NAME: &str = "libkrunfw-tdx.so.5";
+const KRUNFW_NAME: &str = "$ORIGIN/libkrunfw-tdx.so.5";
 #[cfg(target_os = "macos")]
-const KRUNFW_NAME: &str = "libkrunfw.5.dylib";
+const KRUNFW_NAME: &str = "@loader_path/libkrunfw.5.dylib";
 
 #[cfg(feature = "aws-nitro")]
 static KRUN_NITRO_DEBUG: Mutex<bool> = Mutex::new(false);
